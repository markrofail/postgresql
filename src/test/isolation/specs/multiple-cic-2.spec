# Test multiple CREATE INDEX CONCURRENTLY working simultaneously

setup
{
  CREATE TABLE mcic_one (
	id int
  );
}
teardown
{
  DROP TABLE mcic_one;
}

session "s1"
step "s1l"	{ SELECT pg_advisory_lock(281458); }
step "s1i"	{
		CREATE INDEX CONCURRENTLY mcic_one_pkey ON mcic_one (id)
	}
step "s1ul"	{ SELECT pg_advisory_unlock_all(); }
teardown	{ SELECT pg_advisory_unlock_all(); }


session "s2"
step "s2l"	{ SELECT pg_advisory_lock(281458); }
step "s2i"	{
		CREATE INDEX CONCURRENTLY mcic_one_pkey2 ON mcic_one (id)
	}
teardown	{ SELECT pg_advisory_unlock_all(); }


permutation "s1l" "s2l" "s1i" "s1ul" "s2i"
permutation "s1l" "s1i" "s2l" "s1ul" "s2i"
permutation "s1l" "s2i" "s1i" "s1ul"
