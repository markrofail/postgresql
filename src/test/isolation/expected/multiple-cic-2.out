Parsed test spec with 2 sessions

starting permutation: s1l s2l s1i s1ul s2i
step s1l: SELECT pg_advisory_lock(281458);
pg_advisory_lock

               
step s2l: SELECT pg_advisory_lock(281458); <waiting ...>
step s1i: 
		CREATE INDEX CONCURRENTLY mcic_one_pkey ON mcic_one (id)

step s1ul: SELECT pg_advisory_unlock_all();
pg_advisory_unlock_all

               
step s2l: <... completed>
pg_advisory_lock

               
step s2i: 
		CREATE INDEX CONCURRENTLY mcic_one_pkey2 ON mcic_one (id)

pg_advisory_unlock_all

               
pg_advisory_unlock_all

               

starting permutation: s1l s1i s2l s1ul s2i
step s1l: SELECT pg_advisory_lock(281458);
pg_advisory_lock

               
step s1i: 
		CREATE INDEX CONCURRENTLY mcic_one_pkey ON mcic_one (id)

step s2l: SELECT pg_advisory_lock(281458); <waiting ...>
step s1ul: SELECT pg_advisory_unlock_all();
pg_advisory_unlock_all

               
step s2l: <... completed>
pg_advisory_lock

               
step s2i: 
		CREATE INDEX CONCURRENTLY mcic_one_pkey2 ON mcic_one (id)

pg_advisory_unlock_all

               
pg_advisory_unlock_all

               

starting permutation: s1l s2i s1i s1ul
step s1l: SELECT pg_advisory_lock(281458);
pg_advisory_lock

               
step s2i: 
		CREATE INDEX CONCURRENTLY mcic_one_pkey2 ON mcic_one (id)

step s1i: 
		CREATE INDEX CONCURRENTLY mcic_one_pkey ON mcic_one (id)

step s1ul: SELECT pg_advisory_unlock_all();
pg_advisory_unlock_all

               
pg_advisory_unlock_all

               
pg_advisory_unlock_all

               
